File: RemoveDuplicatesFromSortedArray.java
public class Solution {
	public int removeDuplicates(int[] A) {
		if(A.length == 0) {
			return 0;
		}
		int value = A[0];
		int index = 1;
		for(int i = 1; i < A.length; i++) {
			if(A[i] == value) {
				continue;
			} else {
				A[index++] = A[i];
				value = A[i];
			}
		}
		return index;
	}
}

File: RemoveDuplicatesFromSortedArray.cc
class Solution {
    public:
        int removeDuplicates(int A[], int n) {
            if (n == 0) {
                return 0;
            }
            int read_i = 1, write_i = 1;
            while(read_i < n) {
                if (A[read_i] != A[write_i - 1]) {
                    A[write_i++] = A[read_i++];
                } else {
                    read_i++;
                }
            }
            return write_i;
        }
};
